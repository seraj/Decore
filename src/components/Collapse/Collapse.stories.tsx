import { boolean, withKnobs } from "@storybook/addon-knobs";

import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Collapse from "./Collapse";

const stories = storiesOf("Components/Collapse", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => <Collapse isOpen={boolean("isOpen", false)}>content</Collapse>)
);
