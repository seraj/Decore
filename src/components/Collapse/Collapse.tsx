import * as React from "react";
import StyledCollapse from "./styled/StyledCollapse";
import CollapseProps from "./Collapse.props";

const Collapse: React.FC<CollapseProps> = ({ children, ...props }) => (
  <StyledCollapse {...props}>{children}</StyledCollapse>
);

export default Collapse;
