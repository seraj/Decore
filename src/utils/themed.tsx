import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
