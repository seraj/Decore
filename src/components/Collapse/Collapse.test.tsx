import * as React from "react";

import themed from "../../utils/themed";
import Collapse from "./Collapse";

describe("Collapse", () => {
  it("should render correctly", () => {
    const component = themed(<Collapse isOpen={false}>Test</Collapse>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
