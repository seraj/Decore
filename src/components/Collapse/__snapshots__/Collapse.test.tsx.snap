// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should render correctly 1`] = `
<div
  className="collapse"
  style={Object {}}
>
  Test
</div>
`;
